### PR TITLE
Fix OutOfBand parameter and Control class inconsistencies

### LINKED ISSUE #11789: Format *Control class inconsistencies
Is there a reason why the following Control properties are set to either internal setter or only getter when the other Control classes have all properties public setters?

# ListControl

https://github.com/PowerShell/PowerShell/blob/a5a97a593984a8f4ace4cbc19d11429170859503/src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_List.cs#L121

# WideControl
https://github.com/PowerShell/PowerShell/blob/d80154430d600a52a1d1e3a9f3ecf032b36185d4/src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_Wide.cs#L69

https://github.com/PowerShell/PowerShell/blob/d80154430d600a52a1d1e3a9f3ecf032b36185d4/src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_Wide.cs#L75

# CustomControl

Does not have a public constructor even though you can create a default object and assign properties manually.

https://github.com/PowerShell/PowerShell/blob/d80154430d600a52a1d1e3a9f3ecf032b36185d4/src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_Complex.cs#L67-L70

```powershell
C:\> [System.Management.Automation.CustomControl]::Create().EndControl() | gm


   TypeName: System.Management.Automation.CustomControl

Name        MemberType Definition
----        ---------- ----------
Equals      Method     bool Equals(System.Object obj)
GetHashCode Method     int GetHashCode()
GetType     Method     type GetType()
ToString    Method     string ToString()
Entries     Property   System.Collections.Generic.List[System.Management.Automation.CustomControlEntry] Entries {get;set;}
GroupBy     Property   System.Management.Automation.PSControlGroupBy GroupBy {get;set;}
OutOfBand   Property   bool OutOfBand {get;set;}
```

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.0.0-rc.2
PSEdition                      Core
GitCommitId                    7.0.0-rc.2
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by ThomasNieto**
# PR Summary

This PI fixes #11789 and #11788. This allows a module author to create a view definition and sub objects without using the control builder.

## PR Context

The PR makes the following changes:

### CustomControl Class

* Make default constructor public

### ListControl/WideControl Class

* Make setters public on class and property classes
* Fix OutOfBand parameter on `Create` method

### TableControl Class

* Make setters public on property classes

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by ThomasNieto** [comment]
@iSazonov do you think this PR could make it into 7?

**Comment by iSazonov** [comment]
@ThomasNieto 7.0 release will be in days. All new commits will be only in 7.1 (exception critical and security fixes).

As for the PR it makes changes in public API that should be improved (perhaps by PowerShell Committee) and pass security review. Also without previous discussions in issues and without shared business scenarios the review can take many time.

**Comment by iSazonov** [comment]
@daxian-dbw @JamesWTruher @SteveL-MSFT Please review the PR.

**Comment by ThomasNieto** [comment]
Can I please get a review on this?

File: src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_List.cs [hunk 1]
```diff
@@ -201,7 +201,7 @@ internal override bool CompatibleWithOldPowerShell()
     public sealed class ListControlEntry
     {
         /// <summary>List of items in the entry</summary>
-        public List<ListControlEntryItem> Items { get; internal set; }
+        public List<ListControlEntryItem> Items { get; set; }
```
Comment (vexx32) [inline src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_List.cs:204]: The comment here should be updated per the standard format (`Gets or sets ...`)

Comment (vexx32) [reply src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_List.cs:204]: There are a handful of other cases where we're making setters public; we should do the same for all of them.

**Review by vexx32 — APPROVED** [review]
Overall looks good to me; we should update the XML comments for new public APIs and make sure the comments are in line with our standard style.

This will almost certainly need to be checked over to ensure the API changes don't expose any additional concerns since we're making some additional things public here, but the changes themselves seem sound. Nice work! 🙂

**Comment by iSazonov** [comment]
@ThomasNieto Please rebase to get latest CI updates.

**Comment by ThomasNieto** [comment]
Ugh, doesn't look like I did that right.

**Comment by vexx32** [comment]
Hmm, looks like you might have done a sync after a rebase? You'll probably need to re-do the rebase and then force-push once you clear out the excess commits. You can use `git rebase -i upstream/master` if you need to pare out some of the extraneous commits (assuming you have the main PowerShell repo registered as a remote named `upstream` in your local git configuration).

**Comment by ThomasNieto** [comment]
I'm not sure how to do that. Would it be easier to just create a new PR?

**Comment by iSazonov** [comment]
> Would it be easier to just create a new PR?

Feel free to open new PR.

**Comment by ThomasNieto** [comment]
I'll open a new PR later today.

**Pull request closed without merging**